Period on period (#199)
Data-Simply/pyretailscience: docs/analysis_modules.md

)
```

### Period-on-Period Plot

![Period-on-Period Plot](assets/images/analysis_modules/plots/period_on_period.svg)

Period-on-period plots help compare the same metric across two or more time intervals,
all aligned to a common starting point. This is useful when you want to:

- Compare different promotional weeks
- Analyze performance across multiple holiday seasons
- Benchmark key metrics across repeated events (e.g., monthly product launches)

Each period is overlaid on the same plot, allowing for easy visual comparison of trends across intervals.

> **Note:** Dates are automatically realigned to a reference start date,

> so all lines start at the same x=0 point, regardless of calendar time.

Reviewer: This needs to be changed to mention that it starts with the same start year rather than start date

)
```

### Period-on-Period Plot

![Period-on-Period Plot](assets/images/analysis_modules/plots/period_on_period.svg)

> Period-on-period plots help compare the same metric across multiple time intervals, all aligned to a common

Reviewer: I just wanted to call out specifically that it does more than 2 intervals
```suggestion
Period-on-period plots help compare the same metric across two or more time intervals, all aligned to a common
```

starting point. This is useful when you want to:

- Compare different promotional weeks
- Analyze performance across multiple holiday seasons
- Benchmark key metrics across repeated events (e.g., monthly product launches)

Each period is overlaid on the same plot, allowing for easy visual comparison of trends across intervals.

> **Note:** Dates are automatically realigned to a reference start date, so all lines start at the same x=0 point, regardless of calendar time.

#### Example

```python
import pandas as pd
from pyretailscience.plots.period_on_period import overlapping_periods

> periods = [

Reviewer: Can you just show 2 periods please to simplify the example

    ("2022-01-01", "2022-04-01"),
    ("2023-01-01", "2023-04-01"),
    ("2024-01-01", "2024-04-01"),
]
data = {
    'date': [
        '2022-01-02', '2022-01-09', '2022-01-16', '2022-01-23', '2022-01-30',
        '2022-02-06', '2022-02-13', '2022-02-20', '2022-02-27', '2022-03-06',
        '2022-03-13', '2022-03-20', '2022-03-27',
        '2023-01-03', '2023-01-08', '2023-01-15', '2023-01-22', '2023-01-29',
        '2023-02-05', '2023-02-12', '2023-02-19', '2023-02-26', '2023-03-05',
        '2023-03-12', '2023-03-19', '2023-03-26',
        '2024-01-07', '2024-01-14', '2024-01-21', '2024-01-28', '2024-02-04',
        '2024-02-11', '2024-02-18', '2024-02-25', '2024-03-03', '2024-03-10',
        '2024-03-17', '2024-03-24', '2024-03-31'
    ],
    'sales': [
        1024, 1199, 1214, 1295, 1249, 1194, 988, 973, 1029, 910, 952, 976, 1099,
        1195, 1316, 1317, 1361, 1403, 1240, 1164, 1053, 984, 1051, 1079, 1141, 1169,
        1337, 1323, 1497, 1349, 1485, 1288, 1229, 1267, 1103, 1048, 1049, 1230, 1282
    ]
}

df = pd.DataFrame(data)

overlapping_periods(
    df=df,
    x_col="date",
    y_col="sales",
    periods=periods,

>     x_label="date",

Reviewer: Typically I don't add an x axis label if it's a date axis. It just makes it a bit cleaner
```suggestion
```

>     y_label="sales",

Reviewer: Let's capitalize the "S" to make it a little more presentable
```suggestion
    y_label="Sales",
```
